documentation/adr/0010-use-mex-for-pyHorace-python-calls.md

```diff
@@ -0,0 +1,56 @@
+[<-previous](0009-brille-integration.md) | next->
+
+# 10. pyHorace Matlab code should use a mex interface to call Python functions
+
+Date: 2020-Dec-02
+
+## Status
+
+Proposed
+
+
+## Context
+
+`pyHorace` ([prototype](https://github.com/mducle/hugo)) is the proposed Python module for PACE
+which packages the Matlab Horace/Herbert codes as a "compiled" library.
+The Matlab code within this library needs to interact with Python code in the host.
+As detailed in [this design document](../../01_pace_python_high_level_discussion.md#-calling-user-defined-functions)
+there are two main methods for the compiled Matlab code to call a Python function:
+
+1. Using a `mex` file.
+2. Using a "reverse communications interface" (RCI).
+
+In the `mex` method, a reference to the Python function is given to the `mex` file which uses the CPython
+API to call that function (in Python) and then automatically converts (copies) the results to Matlab (`Array`) format.
+In the RCI approach, the standard Matlab-Python interface is used but is restricted to specific use of `multifit`.
+This is because the standard Matlab-Python interface does not allow the compiled Matlab code to call Python functions
+directly, it only allows the Python interpreter to call the compiled Matlab functions.
+Thus, to allow `multifit` to work, the Matlab `multifit` class needs to be refactored to allow it to _receive_ the residuals
```

Comment:
that is, receive the residuals from the python code which it is calling. Again it may be good to be a bit more explicit in the wording.